plugins/modules/change_request_task.py

```diff
@@ -0,0 +1,434 @@
+
+
+def validate_params(params, change_request=None):
+    missing = []
+    if params["state"] == "closed":
+        missing.extend(
+            validation.missing_from_params_and_remote(
+                ("close_code", "close_notes"), params, change_request
+            )
+        )
+
+    if params["on_hold"]:
+        compatible, state = validation.check_value_incompatibility(
+            ("pending", "canceled", "closed"), "state", params, change_request
+        )
+        if not compatible:
+            raise errors.ServiceNowError(
+                'Cannot put a task in state "{}" on hold'.format(state)
+            )
+    elif (change_request or {}).get("on_hold"):
+        compatible, state = validation.check_value_incompatibility(
+            ("pending",), "state", params, change_request
+        )
+        if not compatible:
+            raise errors.ServiceNowError(
+                'Cannot change task state to "{}" when on hold'.format(state)
+            )
+
+    # Description must be set
+    missing.extend(
+        validation.missing_from_params_and_remote(
+            ("short_description", "description"), params, change_request
+        )
+    )
+
+    if missing:
+        raise errors.ServiceNowError(
+            "Missing required parameters {0}".format(", ".join(missing))
+        )
+
+
+def ensure_present(module, table_client):
+    mapper = utils.PayloadMapper(PAYLOAD_FIELDS_MAPPING, module.warn)
+    query = utils.filter_dict(module.params, "sys_id", "number")
+    payload = build_payload(module, table_client)
+
+    if not query:
+        # User did not specify an existing change task, so we need to create a new one.
+        validate_params(module.params)
+        new = mapper.to_ansible(
+            table_client.create_record(
+                "change_task", mapper.to_snow(payload), module.check_mode
+            )
+        )
+        return True, new, dict(before=None, after=new)
+
+    old = mapper.to_ansible(
+        table_client.get_record("change_task", query, must_exist=True)
+    )
+    if is_superset_with_date(old, payload):
+        # No change in parameters we are interested in - nothing to do.
+        return False, old, dict(before=old, after=old)
+
+    validate_params(module.params, old)
+    new = mapper.to_ansible(
+        table_client.update_record(
+            "change_task",
+            mapper.to_snow(old),
+            mapper.to_snow(payload),
+            module.check_mode,
+        )
+    )
+    return True, new, dict(before=old, after=new)
+
+
+def is_superset_with_date(superset, candidate):
+    for k, v in candidate.items():
+        if k not in superset or superset[k] != v:
+            if not (
+                (k == "planned_start_date" or k == "planned_end_date")
+                and (str(superset[k]) != "" and str(v) != "")
+                and str(superset[k]).replace("T", " ") == str(v).replace("T", " ")
+            ):
+                return False
+    return True
```

Comment:
This feels a bit awkward. Since we are only interested in adjusting the result in case any of the two date fields are different in representation but actually represent the same point in time, we should probably do something like this:

```suggestion
def is_superset_with_date(superset, candidate):
    datetime_fields = ("planned_start_date", "planned_end_date")
    
    for k in datetime_fields:
        if k in candidate and superset.get(k, "").replace("T", " ") != candidate[k].replace("T", " "):
            return False
    
    return utils.is_superset(superset, dict((k, v) for k, v in candidate.items() if k not in datetime_fields))
```